bundles/org.openhab.binding.amberelectric/src/main/java/org/openhab/binding/amberelectric/internal/AmberElectricHandler.java

```diff
@@ -0,0 +1,169 @@
+
+    public AmberElectricHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("This binding is read only");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AmberElectricConfiguration.class);
+        if (config.apiKey.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.no-api-key");
+            return;
+        }
+
+        webTargets = new AmberElectricWebTargets();
+        updateStatus(ThingStatus.UNKNOWN);
+        refreshInterval = config.refresh;
+        nmi = config.nmi;
+        apiKey = config.apiKey;
+
+        schedulePoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        logger.debug("Scheduling poll every {} s", refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() throws IOException {
+
+        try {
+            if (siteID.isEmpty()) {
+                Sites sites = webTargets.getSites(apiKey, nmi);
+                // add error handling
+                siteID = sites.siteid;
+                Configuration configuration = editConfiguration();
+                configuration.put("nmi", sites.nmi);
+                updateConfiguration(configuration);
+                logger.debug("Detected amber siteid is {}, for nmi {}", sites.siteid, sites.nmi);
+            }
+
+            CurrentPrices currentPrices = webTargets.getCurrentPrices(siteID, apiKey);
+            final String electricityUnit = " AUD/kWh";
+
+            updateStatus(ThingStatus.ONLINE);
+            Unit<?> unit = CurrencyUnits.getInstance().getUnit("AUD");
+            if (unit == null) {
+                logger.trace("Currency AUD is unknown, falling back to DecimalType");
+                updateState(AmberElectricBindingConstants.CHANNEL_ELECTRICITY_PRICE,
+                        new DecimalType(currentPrices.elecPerKwh));
```

Comment:
Is there a reason for providing values in cent for this case, and otherwise in $? That makes it even more unpredictable what to expect. If a raw `Number` item would be linked, you could otherwise simply ignore the currency unit, and the values would be the same either way.

Reply:
Nope,  it's a mistake.

Reply:
Fixed.